src/autoload/Multiplayer.gd

```diff
@@ -28,17 +30,68 @@ func _ready():
 	assert(not error)
 	error = get_tree().connect("server_disconnected", self, "_server_disconnected")
 	assert(not error)
+	if OS.has_feature("Server") or "--dedicated" in OS.get_cmdline_args():
+		run_dedicated_server()
+
+
+func run_dedicated_server() -> void:
+	dedicated_server = true
+	print("Starting dedicated server")
+	var args := OS.get_cmdline_args()
+	for i in range(args.size()):
+		if args[i] == "--port":
+			if i == args.size() - 1:
+				print("Error, please specify a port number after --port")
+				get_tree().quit(1)
+				return
+			var port_arg := args[i + 1]
+			if not port_arg.is_valid_integer():
+				print("Error, \"%s\" is not a valid integer" % port_arg)
+				get_tree().quit(1)
+				return
+			var new_port := int(port_arg)
+			if new_port < 0 or new_port > 65535:
+				print("Error, port must be between 0 and 65535, found %d" % new_port)
+				get_tree().quit(1)
+				return
+			Global.config.port = new_port
+			break
+	var error := host_server()
+	if error:
+		print("Error, unable to host a server")
+		get_tree().quit(1)
+		return
+	print("Hosting a dedicated server on port %d" % Global.config.port)
+	get_tree().change_scene("res://src/states/Lobby.tscn")
+
+
+# Attempts to create a server and sets the network peer if successful
+func host_server() -> int:
+	var peer := NetworkedMultiplayerENet.new()
+	var error := peer.create_server(Global.config.port, 4)
```

Comment:
I'd change max_clients to 8, or else have this also be a cmd line arg

Reply:
Good idea, didn't even know that was a variable we had

Reply:
Yeah I think it makes sense to have servers be able to limit the number of clients. Live maybe you want a server just for 1v1s or something